### PR TITLE
Create tasks and jobs to create BMC username and password

### PR DESCRIPTION
**Pull request by tannoa2**
The second task updates the mongoDB with new obmSettings.

### CODE REVIEW

**Comment by tannoa2** [comment]
@RackHD/corecommitters @uppalk1 @keedya @zyoung51 @jlongever

**Comment by jlongever** [comment]
test this please

**Comment by benbp** [comment]
:+1:

**Comment by zyoung51** [comment]
+1

File: lib/task-data/tasks/create-ipmi-obm-settings.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2015, EMC, Inc.
```
Comment (yyscamper) [inline lib/task-data/tasks/create-ipmi-obm-settings.js]: Copyright 2016

File: spec/lib/task-data/tasks/create-ipmi-obm-settings-spec.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+    base.before(function (context) {
+        context.taskdefinition = helper.require('/lib/task-data/tasks/create-ipmi-obm-settings.js');
+    });
+
+    describe('task-data', function () {
+        base.examples();
+    });
+
+});
```
Comment (yyscamper) [inline spec/lib/task-data/tasks/create-ipmi-obm-settings-spec.js]: need a newline symbol at end of file

**Comment by yyscamper** [comment]
:+1: after copyright and newline fix.

File: spec/lib/jobs/generate-bmc-password-spec.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+        context.Jobclass = helper.injector.get('Job.BMC.Password');
+    });
+
+    describe('Base', function () {
+        base.examples();
+    });
+
+    describe("create-bmc-password-job", function() {
+        it('invoke a run function');
```
Comment (jlongever) [inline spec/lib/jobs/generate-bmc-password-spec.js]: Should there be a test case defined here?

**Comment by jlongever** [comment]
Looks like the file permissions were changed 0644 -> 0755.  Should probably keep them set to 0644.
Otherwise +1.

**Comment by keedya** [comment]
my only comment is why we have indentation changes in linux-command job?
otherwise :+1:

File: lib/jobs/generate-bmc-password.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    util.inherits(BmcPasswordJob, BaseJob);
+
+    /**
+     * @memberOf BmcPasswordJob
+     */
+    BmcPasswordJob.prototype._run = function _run() {
+        var self = this;
+
+        var token = crypto.randomBytes(4).toString('hex');
+        this.context.password = token;
```
Comment (zyoung51) [inline lib/jobs/generate-bmc-password.js]: The context.password should be kept if it was set?
this.context.password = this.context.password || token;

**Comment by keedya** [comment]
:+1:

**Comment by zyoung51** [comment]
+1

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by tannoa2** [comment]
test this please

**Comment by jlongever** [comment]
+1

**Pull request merged**